### PR TITLE
Upgrade to zuul 1.2.2

### PR DESCRIPTION
**Pull request by spencergibb**
/cc @dsyer @marcingrzejszczak 

fixes gh-1091
closes gh-958

### CODE REVIEW

File: spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendErrorFilter.java [hunk 1]
```diff
@@ -76,6 +77,8 @@ public Object run() {
 				request.setAttribute("javax.servlet.error.message", message);
 			}
 
+			//unwrap request if needed, otherwise tomcat hangs in endless loop
```
Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendErrorFilter.java]: This is what was causing the endless while loop.

Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendErrorFilter.java]: Even after removing the send 404 to `/error` I still need to unwrap. I'm a bit concerned and wonder if anyone can comment.

The problem occurs in [`ApplicationDispatcher`]( https://github.com/apache/tomcat/blob/TOMCAT_8_0_0/java/org/apache/catalina/core/ApplicationDispatcher.java#L907-L908).

```
if (!(current instanceof ServletRequestWrapper))
                break;
```
The request is always a `ServletRequestWrapper` unless I unwrap it.

@markt-asf @wilkinsona @philwebb @snicoll any thoughts?

Comment (markt-asf) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendErrorFilter.java]: @spencergibb  Source code for the `ServletRequestWrapper` being used at that point?

Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendErrorFilter.java]: @markt-asf https://github.com/Netflix/zuul/blob/1.x/zuul-core/src/main/java/com/netflix/zuul/http/HttpServletRequestWrapper.java

Comment (markt-asf) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendErrorFilter.java]: And there is the problem. getRequest() doesn't return the request that is wrapped by the current wrapper (i.e. req) it returns a new, wrapped version of req. That is a bug in zuul.

Comment (spencergibb) [inline spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/zuul/filters/post/SendErrorFilter.java]: Thanks @markt-asf!

**Pull request merged**